frontend/src/presentation/pages/QuestionsPage.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState, useEffect } from "react";
+import { Row, Col, Breadcrumb, Button, Spin, Alert } from "antd";
+import { QuestionList } from "../components/QuestionList";
+import { QuestionDetail } from "../components/QuestionDetail";
+import { LandingComponent } from "../components/LandingComponent";
+import { Question } from "../../domain/entities/Question";
+import { questionUseCases } from "../../domain/usecases/QuestionUseCases";
+import styles from "./QuestionsPage.module.css";
+import { ROUTES, ERRORS } from "presentation/utils/constants";
+import { handleError } from "presentation/utils/errorHandler";
+import { AddQuestionButton } from "presentation/components/buttons/AddQuestionButton";
+
+const LAST_SELECTED_QUESTION_KEY = "lastSelectedQuestionId";
+
+const QuestionsPage: React.FC = () => {
+    const [questions, setQuestions] = useState<Question[]>([]);
+    const [selectedQuestionId, setSelectedQuestionId] = useState<string | null>(
+        null
+    );
+    const [selectedQuestion, setSelectedQuestion] = useState<Question | null>(
+        null
+    );
+    const [isLoading, setIsLoading] = useState(true);
+    const [isQuestionLoading, setIsQuestionLoading] = useState(false);
+    const [error, setError] = useState<string | null>(null);
+
+    useEffect(() => {
+        const fetchQuestions = async () => {
+            setIsLoading(true);
+            try {
+                const fetchedQuestions = await questionUseCases.getAllQuestions();
+                setQuestions(fetchedQuestions);
+            } catch (err) {
+                setError(handleError(err, ERRORS.FAILED_TO_LOAD_QUESTIONS));
+            } finally {
+                setIsLoading(false);
+            }
+        };
+
+        fetchQuestions();
+
+        //Should we use slugs/query params instead of this?
+        const lastSelectedQuestionId = localStorage.getItem(
+            LAST_SELECTED_QUESTION_KEY
+        );
+
+        if (lastSelectedQuestionId) {
+            setSelectedQuestionId(lastSelectedQuestionId);
+        }
+    }, []);
+
+    useEffect(() => {
+        const fetchSelectedQuestion = async () => {
+            if (selectedQuestionId) {
+                setIsQuestionLoading(true);
+                try {
+                    const question = await questionUseCases.getQuestion(
+                        selectedQuestionId
+                    );
+                    setSelectedQuestion(question);
+                    localStorage.setItem(LAST_SELECTED_QUESTION_KEY, selectedQuestionId);
```

Comment:
local storage is unncessary here and try not to use the local storage for state management.